src/main/java/cart/controller/ProductController.java

```diff
@@ -0,0 +1,50 @@
+package cart.controller;
+
+import cart.domain.Product;
+import cart.dto.request.ProductSaveRequest;
+import cart.dto.request.ProductUpdateRequest;
+import cart.service.ProductService;
+import java.util.List;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+public class ProductController {
+
+    private final ProductService productService;
+
+    public ProductController(ProductService productService) {
+        this.productService = productService;
+    }
+
+    @GetMapping("/product")
+    public ResponseEntity<List<Product>> findAllProducts() {
+        List<Product> products = productService.findAllProducts();
+        return ResponseEntity.ok().body(products);
+    }
+
+    @PostMapping("/product")
+    public ResponseEntity<String> saveProduct(@RequestBody ProductSaveRequest productSaveRequest) {
+        productService.saveProduct(productSaveRequest);
+        return ResponseEntity.status(HttpStatus.CREATED).build();
+    }
+
+    @PutMapping("/product")
```

Comment:
```suggestion
    @PutMapping("/products/{productId}")
```
requestBody보다 url에 id를 넣는건 어떻게 생각하세요?
url path는 리소스 경로의 의미를 가지고 있어서요.

Reply:
이 방식이 더 일반적으로 제안되는 방법이라는데 동의합니다! 일단 해당 방식대로 수정하겠습니다.

그런데 궁금한 점은 productId도 보안적인 관점에서는 노출되지 않는 게 나을 듯 한데,
requestBody에 넣거나, 혹은 header에 넣는 방법을 택할 수도 있지 않나요?
product의 id만으로는 크리티컬한 노출이 아니고 오히려 접근할 자원을 명확하게 보여준다는 점에서 이점으로 보는 것인지? 이 방식의 장점이 무엇일지 궁금합니다!

Reply:
보안에 대해서 id가 노출되는 것은 위험할 수도 있겠지만 식별할 수 있는 값은 필요하지 않을까요? 다른 header나 requestbody에 넣는다면 보안 측면에서 달라질까요?

restfulAPI는 객체지향처럼 사람마다 설계 기준이 다를 수 있을 것 같습니다.
request로 받아 올 때, 각각에 대해 어떤 기준으로 설정할지 기준을 세워보면 좋을 것 같아요!
- header
- path variable
- request body
- request parameters


restfulAPI에 대해 아래 글을 참고해봐도 좋을 듯 싶습니다! :) 

https://meetup.nhncloud.com/posts/92